devU-api/src/utils/fileUpload.utils.ts

```diff
@@ -0,0 +1,33 @@
+import { BucketNames} from '../fileStorage'
+import { fileUploadTypes } from '../../devu-shared-modules'
+import  crypto from 'crypto'
+
+/*
+  * generateFilename currently only use sha256 to hash the originalName with a timestamp
+  * This can be updated to use a different hashing algorithm in the future
+  * Marking for discussion
+ */
+export function generateFilename(originalName: string): string {
+  const hash = crypto.createHash('sha256')
+  const timestamp = Date.now();
+  hash.update(originalName+timestamp)
+  return hash.digest('hex')
+}
+
+
+/*
+  This function is deciding which bucket to upload the file
+  however, it only specifies the bucket for graderFile and makefileFile
+  It can be updated to support multiple buckets in the future
+
+  Marking for discussion
+
+ */
```

Comment:
In discussion. This will be changed in the near future. Current idea is to have a bucket for each user and a bucket for each course. Everyone has access to their bucket and is used for student uploads. Course buckets for graders and can be accessed by instructors and TAs of that course. Instructors can add files to a student's bucket to create a submission on their behalf. Each student submission should know which course it was submitted to.